src/javascripts/helpers/validationForm.js

```diff
@@ -48,30 +48,38 @@ export const validateTransaction = (data) => {
   const transactionFields = [
     {
       field: 'category',
-      requiredMessage: MESSAGE.CATEGORY_REQUIRED
+      requiredMessage: MESSAGE.CATEGORY_REQUIRED,
+      undefinedMessage: MESSAGE.CATEGORY_UNDEFINED
     },
     {
       field: 'date',
-      requiredMessage: MESSAGE.DATE_REQUIRED
+      requiredMessage: MESSAGE.DATE_REQUIRED,
+      undefinedMessage: MESSAGE.DATE_UNDEFINED
     },
     {
       field: 'outflow',
-      requiredMessage: MESSAGE.AMOUNT_REQUIRED
+      requiredMessage: MESSAGE.AMOUNT_REQUIRED,
+      undefinedMessage: MESSAGE.AMOUNT_UNDEFINED
     },
     {
       field: 'note',
-      requiredMessage: MESSAGE.NOTE_REQUIRED
+      requiredMessage: MESSAGE.NOTE_REQUIRED,
+      undefinedMessage: MESSAGE.NOTE_UNDEFINED
     },
   ]
 
-  for(const key in data){
-    const value = data[key];
-    const transactionField = transactionFields.find((field) => field.field === key)
+  for(const field of transactionFields){
+    const value = data[field.field];
 
-    if(value.trim() === ''){
+    if (value === undefined){
       fieldCheck.push({
-        field: key,
-        message: transactionField.requiredMessage
+        field: field.field,
+        message: field.undefinedMessage
+      })
+    } else if(value.trim() === ''){
```

Comment:
Did we need if here?

Reply:
I think we need if here because `if () .... else if ()....` is used when the conditions in the first if are true, then only execute the statements in the first if and stop. In our case, if data is null then we will only check for null and stop there, not checking the following cases because data is null